pkg/kubelet/kubelet.go

```diff
@@ -1493,7 +1491,25 @@ func (kl *Kubelet) initializeRuntimeDependentModules() {
 func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 	ctx := context.Background()
 	if kl.logServer == nil {
-		kl.logServer = http.StripPrefix("/logs/", http.FileServer(http.Dir("/var/log/")))
+		file := http.FileServer(http.Dir("/var/log/"))
+		if utilfeature.DefaultFeatureGate.Enabled(features.NodeLogQuery) {
+			kl.logServer = http.StripPrefix("/logs/", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+				query := req.URL.Query()
```

Comment:
this is a general comment, but there's a lot of repeated lookup of query parameters by name in validateNodeLogQueryCombination / parseQueryForServices / newJournalArgsFromURL ... those are vulnerable to typos and make it hard to keep track of when we've extracted / validated data in the request params

can we do a url.Values → NodeLogOptions extraction / validation as a pre-step, which does:
* well-formedness (values must parse if specified)
* safety (values must pass safe string)
* semantically valid (untilTime must be after sinceTime if both are specified, tailLines must be positive if specified, etc)

then work with the typed options struct inside journal.ServeHTTP?